src/test/scala/io/iohk/ethereum/jsonrpc/EthServiceSpec.scala

```diff
@@ -489,69 +450,8 @@ class EthServiceSpec extends FlatSpec with Matchers with ScalaFutures with MockF
-
-    ethService.submitHashRate(SubmitHashRateRequest(12, id)).futureValue shouldEqual Right(SubmitHashRateResponse(true))
-    ethService.submitHashRate(SubmitHashRateRequest(rate, id)).futureValue shouldEqual Right(SubmitHashRateResponse(true))
-
-    val response = ethService.getHashRate(GetHashRateRequest())
-    response.futureValue shouldEqual Right(GetHashRateResponse(rate))
-  }
-
-  it should "combine hashrates from many miners and remove timed out rates" in new TestSetup {
-    val rate: BigInt = 42
-    val id1 = ByteString("id1")
-    val id2 = ByteString("id2")
-
-    ethService.submitHashRate(SubmitHashRateRequest(rate, id1)).futureValue shouldEqual Right(SubmitHashRateResponse(true))
-    Thread.sleep(miningConfig.activeTimeout.toMillis / 2)
-    ethService.submitHashRate(SubmitHashRateRequest(rate, id2)).futureValue shouldEqual Right(SubmitHashRateResponse(true))
-
-    val response1 = ethService.getHashRate(GetHashRateRequest())
-    response1.futureValue shouldEqual Right(GetHashRateResponse(rate * 2))
-
-    Thread.sleep(miningConfig.activeTimeout.toMillis / 2)
-    val response2 = ethService.getHashRate(GetHashRateRequest())
-    response2.futureValue shouldEqual Right(GetHashRateResponse(rate))
-  }
-
-  it should "return if node is mining base on getWork" in new TestSetup {
-    ethService.getMining(GetMiningRequest()).futureValue shouldEqual Right(GetMiningResponse(false))
-
-    (blockGenerator.generateBlockForMining _).expects(parentBlock, *, *, *).returning(Right(PendingBlock(block, Nil)))
-    blockchain.save(parentBlock)
-    ethService.getWork(GetWorkRequest())
-
-    val response = ethService.getMining(GetMiningRequest())
-
-    response.futureValue shouldEqual Right(GetMiningResponse(true))
-  }
-
-  it should "return if node is mining base on submitWork" in new TestSetup {
-    ethService.getMining(GetMiningRequest()).futureValue shouldEqual Right(GetMiningResponse(false))
-
-    (blockGenerator.getPrepared _).expects(*).returning(Some(PendingBlock(block, Nil)))
-    (appStateStorage.getBestBlockNumber _).expects().returning(0)
-    ethService.submitWork(SubmitWorkRequest(ByteString("nonce"), ByteString(Hex.decode("01" * 32)), ByteString(Hex.decode("01" * 32))))
-
-    val response = ethService.getMining(GetMiningRequest())
-
-    response.futureValue shouldEqual Right(GetMiningResponse(true))
-  }
-
-  it should "return if node is mining base on submitHashRate" in new TestSetup {
-    ethService.getMining(GetMiningRequest()).futureValue shouldEqual Right(GetMiningResponse(false))
-
-    ethService.submitHashRate(SubmitHashRateRequest(42, ByteString("id")))
-
-    val response = ethService.getMining(GetMiningRequest())
-
-    response.futureValue shouldEqual Right(GetMiningResponse(true))
-  }
-
-  it should "return if node is mining after time out" in new TestSetup {
+  // TODO: Check if this test could be fixed it
+  /*it should "return if node is mining after time out" in new TestSetup {
```

Comment:
We should probably remove this if `eth_mining` is removed, right? If `eth_mining` is set to return always true, maybe we can replace this test taking into account that.

Reply:
Deleted then!